### PR TITLE
Initial Elevio integration added

### PR DESCRIPTION
**Pull request by duellsy**
Allows users to install elevio through Segment.

Uses the identify method to pass user information through to elevio to display relevant help information to the user.

Just let me know if there's anything else required.

### CODE REVIEW

File: lib/elevio/index.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Elevio.prototype.identify = function(identify){
+    var name;
+    if (identify.firstName() !== undefined) {
+        name = identify.firstName();
+        if (identify.lastName() !== undefined) {
+            name += ' ' + identify.lastName();
+        }
+    } else if (identify.name() !== undefined) {
+        name = identify.name();
+    }
```
Comment (sperand-io) [inline lib/elevio/index.js]: as this is already handled by the [identify facade](https://github.com/segmentio/facade/blob/master/lib/identify.js#L115-L164),  `var name = identify.name()` should suffice, no?

File: lib/elevio/index.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  var email = identify.email();
+  var plan = identify.traits().plan;
```
Comment (sperand-io) [inline lib/elevio/index.js]: best practice is to use `identify.proxy('traits.plan')` here, as that will afford case insensitivity

**Comment by sperand-io** [comment]
two comments inline — awesome first pass! thanks!

File: lib/elevio/index.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  });
+}
+
+Elevio.prototype.load = function(callback){
+    window._elev = window._elev || {};(function(){var i,e;i=document.createElement("script"),i.type='text/javascript';i.async=1,i.src="https://static.elev.io/js/v3.js",e=document.getElementsByTagName("script")[0],e.parentNode.insertBefore(i,e);})();
+    // window._elev = window._elev || {};(function(){var i,e;i=document.createElement("script"),i.type='text/javascript';i.async=1,i.src="http://widget.elev.dev:8000/scripts/v3.js",e=document.getElementsByTagName("script")[0],e.parentNode.insertBefore(i,e);})();
+    _elev.account_id = this.options.accountId;
+    _elev.segment = true;
+    callback();
+};
```
Comment (sperand-io) [inline lib/elevio/index.js]: looks like we can we move this into the initialize method.

you can declare your script with the `.tag()` static method on the integration constructor, and then `this.load()` will download the tag accordingly.

With that in mind, can we give this a try in lieu of overriding the load method? (I haven't tested but should work :)

```
/**
 * Expose `Elevio` integration.
 */

var Elevio = module.exports = integration('Elevio')
  .assumesPageview()
  .option('accountId', '')
  .global('_elev')
  .tag('<script src="https://static.elev.io/js/v3.js">');

/**
 * Initialize elevio.
 *
 * @param {Facade} page
 */

Elevio.prototype.initialize = function(page){
  window._elev = window._elev || {};
  _elev.account_id = this.options.accountId;
  _elev.segment = true;
  this.load(self.ready);
}
```

**Comment by duellsy** [comment]
Awesome, I'll close this pull, make the requested updates and submit a new pull request.
Thanks for the feedback :+1:

**Comment by f2prateek** [comment]
@duellsy feel free to push the changes in this commit directly, makes code reviews easier rather than going back and forth across multiple PRs :)

File: lib/elevio/index.js (hunk 1 above)
Comment (duellsy) [inline lib/elevio/index.js]: Makes sense, I've changed to use the identify method, however it seems Identify.prototype.name returns nothing when no last name is sent

Comment (duellsy) [inline lib/elevio/index.js]: @sperand-io makes sense I've updated to use the identify facede however it seems `Identify.prototype.name` returns nothing when no last name is provided

**Comment by duellsy** [comment]
@f2prateek & @sperand-io updates made, and PR re-opened.
Thanks for the feedback!

**Comment by duellsy** [comment]
Not sure what the error is for in the test on circleci, when I run the tests locally it app passes with flying colours:
`Error loading resource https://static.elev.io/js/v3.js (6). Details: SSL handshake failed`

**Comment by duellsy** [comment]
Looking for some ideas on why there's the SSL issues when tests are run on CircleCI but not locally, any ideas @sperand-io / @f2prateek ?

**Comment by f2prateek** [comment]
@duellsy Thanks! We're looking into the SSL errors.

**Comment by duellsy** [comment]
thanks, looking forward to getting this resolved, got me stumped!

**Comment by yields** [comment]
@duellsy fixed that ssl issue, seems like the tests are now failing because of another issue, can you please rebase? maybe those issues are resolved on master now

**Comment by duellsy** [comment]
@yields on it, I'll work with it and report back. Thanks.

**Comment by duellsy** [comment]
@yields our tests look to work fine, but there's 1 error when running all tests, which is test 1 under #initialize for the Google Analytics integration.

I'm assuming this is unrelated to us?

**Comment by ndhoule** [comment]
@duellsy Yeah, those GA issues are unrelated to you. This is good to go and landed over in https://github.com/segmentio/analytics.js-integrations/pull/638. Thanks for the PR!

**Pull request closed without merging**